test/malli/core_test.cljc

```diff
@@ -997,6 +997,18 @@
                                                (mt/key-transformer
                                                 {:decode #(-> % name (str "_key") keyword)}))))
 
+      (testing "JSON transformer decodes map schema keys"
+        (let [schema [:map
+                      [:a :uuid]
+                      [:b [:enum :x :y :z]]]
```

Comment:
used non-trivial schemas for leaves here just to convince myself that a layer of `(-transform-map-keys -string->keyword)` doesn't mess with recursive decoding